src/server.tsx

```diff
@@ -57,7 +61,11 @@ const startServer = async () => {
 
   server.app.use(async (ctx: any, next: any) => {
     await next();
-    ctx.response.body = await renderSite(ctx.request.url);
+    const { render, is404 } = await renderSite(ctx.request.url);
+    if (is404) {
+      ctx.response.status = 404;
```

Comment:
replace this with

const { render, status } = await renderSite(ctx.request.url);
if (status) {
  ctx.response.status = status.code;
}